packages/solver-sat/src/solver-sat.ts

```diff
@@ -152,23 +152,28 @@ export async function solve(board: schema.Board, maxSolutions: number,
 
 function updateValidCandidatesForSolutions(
     solutions: IdxMap<typeof Geometry.CELL, number>[],
-    validCandidates: IdxMap<Geometry.CELL, Array<number>>,
+    validCandidates: IdxMap<Geometry.CELL, Map<number, number>>,
+	foundSolutions: Record<string, boolean>,
+	grid: Grid,
     size: number,
 ) {
     for (const solution of solutions) {
+		const representation = solutionToString(solution, grid);
+		if(representation in foundSolutions) continue;
+
         for (let cellIndex = 0; cellIndex < size * size; cellIndex++) {
             const value = solution[cellIndex];
             if (undefined === value) continue;
 
-            if (!validCandidates[cellIndex]?.includes(value)) {
-                validCandidates[cellIndex]?.push(value);
-            }
+            const count = validCandidates[cellIndex]?.get(value) || 0;
+            validCandidates[cellIndex]?.set(value, count+1);
         }
+		foundSolutions[representation] = true;
```

Comment:
Looks like this line starts with tabs, which should be spaces instead (I see some other lines in your PR with the same issue)